stablehlo/dialect/TypeInference.cpp

```diff
@@ -166,6 +166,54 @@ LogicalResult verifyBinaryOpQuantizationConstraints(
   return success();
 }
 
+LogicalResult verifyConvolutionDotGeneralCommonQuantizationConstraints(
+    std::optional<Location> location, Type lhsElementType, Type rhsElementType,
+    Type resultElementType) {
+  // convolution_c28
+  if (!rhsElementType.isa<quant::QuantizedType>() ||
+      (lhsElementType.isa<quant::QuantizedType>() !=
+       resultElementType.isa<quant::QuantizedType>())) {
+    return emitOptionalError(
+        location,
+        "rhs should be quantized for quantized operations and "
+        "is_quantized(lhs)=is_quantized(result) should hold");
+  }
+
+  auto rhsQuantType = rhsElementType.dyn_cast<quant::QuantizedType>();
+  if (lhsElementType.isa<quant::QuantizedType>()) {
+    auto lhsQuantType = lhsElementType.dyn_cast<quant::QuantizedType>();
+    auto resultQuantType = resultElementType.dyn_cast<quant::QuantizedType>();
+    // convolution_c31
+    if (lhsQuantType.getStorageType() != rhsQuantType.getStorageType()) {
+      return emitOptionalError(
+          location, "mismatched lhs and rhs quantization storage types");
+    }
+    // convolution_c32
+    if (lhsQuantType.getExpressedType() != rhsQuantType.getExpressedType() ||
+        lhsQuantType.getExpressedType() != resultQuantType.getExpressedType()) {
+      return emitOptionalError(
+          location,
+          "mismatched lhs, rhs and result quantization expressed types");
+    }
+    // convolution_c33
```

Comment:
```
* (C33) If `is_per_tensor_quantized(rhs)`, then
      `is_per_tensor_quantized(result)`.
```

Per the above constraint the result quantization granularity should follow that of rhs **only if rhs is per-tensor**. 
For example, It is possible for the rhs to be per-axis and result to be per-tensor, which will be errored out by the following check.

```
 if (isa<quant::UniformQuantizedType>(rhsQuantType) &&
        !isa<quant::UniformQuantizedType>(resultQuantType)) {
      return emitOptionalError(
          location, "mismatched rhs and result quantization granularity");
    }
```

Reply:
Thanks for catching! Updated.